src/main.cpp

```diff
@@ -471,13 +482,42 @@ DEMUX_PACKET* CInputStreamAdaptive::DemuxRead(void)
 
     if (iSize > 0 && pData)
     {
-      p->dts = static_cast<double>(sr->DTS() + m_session->GetChapterStartTime());
-      p->pts = static_cast<double>(sr->PTS() + m_session->GetChapterStartTime());
+      uint64_t chapterStartTime{m_session->GetChapterStartTime()};
+      p->dts = static_cast<double>(sr->DTS() + chapterStartTime);
+      p->pts = static_cast<double>(sr->PTS() + chapterStartTime);
       p->duration = static_cast<double>(sr->GetDuration());
       p->iStreamId = sr->GetStreamId();
       p->iGroupId = 0;
       p->iSize = iSize;
       memcpy(p->pData, pData, iSize);
+
+      if (sr->GetStreamType() == INPUTSTREAM_TYPE::INPUTSTREAM_TYPE_SUBTITLE &&
```

Comment:
I was thinking that here we could put most of this into the sample reader itself to keep this function generic to all sample readers.
i.e
```
p->pSideData = sr->GetSideData(chapterStartTime); // or even if we can keep the chapter starting time on the sample reader when we construct/initialize it?
if (p->SideData)
  p->iSideDataElems = 1;
```
And do all the rest on sr - thoughts?

Reply:
maybe yes i will try do it

Reply:
done much better less changes

Reply:
Looks great. I was wrong with the idea about setting chapterstarttime on construction - periods and starting times can change during playback.